lib/icinga/apiactions.cpp

```diff
@@ -890,6 +891,32 @@ Dictionary::Ptr ApiActions::ExecuteCommand(const ConfigObject::Ptr& object, cons
 						return ApiActions::CreateResult(202, "Accepted", result);
 					}
 				}
+
+				/* Check if the child zone can access the checkable */
+				if (!zone->CanAccessObject(checkable)) {
```

Comment:
This only checks if the direct child zone can access the object. But the hierarchy is known so it should be possible to check earlier if the zone of `endpointPtr` can access `checkable`. This could also make other checks redundant.